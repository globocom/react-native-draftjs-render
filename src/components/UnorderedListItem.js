/*
 * Copyright (c) 2017, Globo.com (https://github.com/globocom)
 *
 * License: MIT
 */

// @flow

import React from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';

import DraftJsText from '../components/DraftJsText';

import type { UnorderedListItemPropsType } from './types';

const styles = StyleSheet.create({
  unorderedListItemContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  unorderedListItemBullet: {
    width: 5,
    height: 5,
    borderRadius: 5,
    marginRight: 8,
    alignSelf: 'center',
    backgroundColor: 'black',
  },
});

const UnorderedListItem = (props: UnorderedListItemPropsType): any => {
  const unorderedListItemCustomStyleContainer = props.customStyles ?
    props.customStyles.unorderedListItemContainer :
    undefined;

  const unorderedListItemCustomStyleBullet = props.customStyles ?
    props.customStyles.unorderedListItemBullet :
    undefined;

  let marginLeft = 0;
  marginLeft =
      unorderedListItemCustomStyleBullet && unorderedListItemCustomStyleBullet.marginLeft ?
        props.depth * unorderedListItemCustomStyleBullet.marginLeft :
        props.depth * props.defaultMarginLeft;

  return (
    <View style={[styles.unorderedListItemContainer, unorderedListItemCustomStyleContainer]}>
      <View style={[styles.unorderedListItemBullet, unorderedListItemCustomStyleBullet,
        { marginLeft }]}
      />
      <DraftJsText
        {...props}
      />
    </View>);
};

UnorderedListItem.defaultProps = {
  defaultMarginLeft: 8,
  depth: 0,
};

export default UnorderedListItem;
