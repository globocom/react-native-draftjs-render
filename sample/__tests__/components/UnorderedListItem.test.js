/*
 * Copyright (c) 2017, Globo.com (https://github.com/globocom)
 *
 * License: MIT
 */

/* eslint-env jest */

import 'react-native';

import React from 'react';
import renderer from 'react-test-renderer';

import UnorderedListItem from '../../../src/components/UnorderedListItem';

it('renders correctly with a unordered-list-item', () => {
  const text = 'Hello World';
  const tree = renderer.create(<UnorderedListItem
    type="unordered-list-item"
    text={text}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with a unordered-list-item when depth is undefined', () => {
  const text = 'Hello World';
  const tree = renderer.create(<UnorderedListItem
    type="unordered-list-item"
    text={text}
    depth={undefined}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with a unordered-list-item when depth 1', () => {
  const text = 'Hello World';
  const tree = renderer.create(<UnorderedListItem
    type="unordered-list-item"
    text={text}
    depth={1}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders null without a unordered-list-item', () => {
  const tree = renderer.create(<UnorderedListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
it('extends a style with customStyles', () => {
  const text = 'Hello World';
  const customStyles = {
    'unordered-list-item': {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
      marginLeft: 10,
    },
  };

  const tree = renderer.create(<UnorderedListItem
    type="unordered-list-item"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('extends a style with a customStyle.unorderedListItemContainer', () => {
  const text = 'Hello World';
  const customStyles = {
    'unordered-list-item': {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
      marginLeft: 10,
    },
    unorderedListItemContainer: {
      flex: 2,
    },
  };
  const tree = renderer.create(<UnorderedListItem
    type="ordered-list-item"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when unorderedListItemBullet.marginLeft is set', () => {
  const text = 'Hello World';
  const customStyles = {
    'unordered-list-item': {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
      marginLeft: 10,
    },
    unorderedListItemBullet: {
      marginLeft: 14,
    },
  };
  const tree = renderer.create(<UnorderedListItem
    type="unordered-list-item"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
