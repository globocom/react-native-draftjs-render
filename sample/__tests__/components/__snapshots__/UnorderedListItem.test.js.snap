// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extends a style with a customStyle 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 5,
          "height": 5,
          "marginRight": 8,
          "width": 5,
        },
        undefined,
        Object {
          "marginLeft": 0,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        Object {
          "fontSize": 18,
          "fontWeight": "normal",
          "letterSpacing": -0.75,
          "lineHeight": 32,
          "marginLeft": 10,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`renders correctly with a unordered-list-item 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 5,
          "height": 5,
          "marginRight": 8,
          "width": 5,
        },
        undefined,
        Object {
          "marginLeft": 0,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`renders correctly with a unordered-list-item when depth 1 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 5,
          "height": 5,
          "marginRight": 8,
          "width": 5,
        },
        undefined,
        Object {
          "marginLeft": 7,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`renders correctly with a unordered-list-item when depth is undefined 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 5,
          "height": 5,
          "marginRight": 8,
          "width": 5,
        },
        undefined,
        Object {
          "marginLeft": 0,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`renders null without a unordered-list-item 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderRadius": 5,
          "height": 5,
          "marginRight": 8,
          "width": 5,
        },
        undefined,
        Object {
          "marginLeft": 0,
        },
      ]
    }
  />
</View>
`;
